### PR TITLE
(#482) targetsdk 29

### LINKED ISSUE #482: [IMPROVEMENT] Switch targetSdkVersion to 29
There are gonna be some caveats which need to be fixed. The most hard of them - SAF - is already done, though. Probably some stuff related to background execution is left.

### PR DESCRIPTION
**Pull request by K1rakishou**
Closes #482

Didn't really have any problems with it so I assume it's safe to use v29.

### CODE REVIEW

File: Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/ThreadSaveManager.java [hunk 1]
```diff
@@ -133,9 +133,10 @@ private void initRxWorkerQueue() {
         // Just buffer everything in the internal queue when the consumers are slow (and they are
         // always slow because they have to download images, but we check whether a download request
         // is already enqueued so it's okay for us to rely on the buffering)
-        workerQueue.onBackpressureBuffer()
+        workerQueue
                 // Collect all the request over some time
                 .buffer(REQUEST_BUFFERING_TIME_SECONDS, SECONDS)
+                .onBackpressureBuffer()
```
Comment (K1rakishou) [inline Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/ThreadSaveManager.java:139]: Should move this to the other PR where all local thread fixes are.

Comment (K1rakishou) [reply Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/manager/ThreadSaveManager.java:139]: Or I could just do the same in that PR and leave this here.

**Comment by K1rakishou** [comment]
Should merge merged either together with https://github.com/Adamantcheese/Kuroba/pull/514 or after that one, because in this PR we don't start the WatchNotification service as foreground service which will blow up the app right after the start if we have at least one pinned thread.

**Comment by Adamantcheese** [comment]
As I noted in my review of #449, there would be an issue for transferring a downloaded APK copy from the cache to the downloads directory if I were to merge this.

**Comment by K1rakishou** [comment]
So I have just checked (just in case) and the updater works.

**Pull request merged**